docs/Formula-Cookbook.md

```diff
@@ -888,7 +892,7 @@ There are two ways to add `launchd` plists and `systemd` services to a formula,
 
 #### Service block methods
 
-This table lists the options you can set within a `service` block. Only the `run` field is required which indicates what to run.
+This table lists the options you can set within a `service` block. The `run` or `name` field must be defined inside the service block. The `run` field indicates what command to run and is required before using fields other than `name`.
```

Comment:
```suggestion
This table lists the options you can set within a `service` block. The `run` or `name` field must be defined inside the service block. The `run` field indicates what command to run and is required unless a `name` of an existing service is defined.
```

Reply:
The way I read the original the following are all valid:
```rb
service do
  name "test"
  run "a"
  something_else true
end
```
```rb
service do
  name "test"
  something_else true
end
```
```rb
service do
  run "a"
  something_else true
end
```

Reply:
That's right but the following is not valid.

```rb
service do
  name "test"
  something_else true
end
```